sorts/dnf_sort.py

```diff
@@ -0,0 +1,77 @@
+"""
+Pure implementation of Dutch national flag problem(dnf) sort algorithm in Python
+Dutch National Flag algorithm is a popular algorithm originally designed by Edsger Dijkstra.
+It is the most optimal sort for 3 unique values (eg. 0,1,2) in an Array.
+dnf sort can sort an array of n size with [0<=a[i]<=2] at guaranteed O(n) complexity in a single pass.
+
+The flag of the Netherlands consists of three colors: white, red, and blue. 
+The task is to randomly arrange balls of white, red, and blue in such a way that balls of the same color are placed together. 
+For DNF (Dutch National Flag), we sort an array of 0, 1, and 2's in linear time that does not consume any extra space. 
+We have to keep in mind that this algorithm can be implemented only on an array that has three unique elements.
+
+1)Time complexity is O(n).
+2)Space complexity is O(1).
+
+More info on: https://en.wikipedia.org/wiki/Dutch_national_flag_problem
+
+For doctests run following command:
+python -m doctest -v dnf_sort.py
+or
+python3 -m doctest -v dnf_sort.py
+
+For manual testing run:
+python dnf_sort.py
+
+"""
+
+
+# Python program to sort an array with
+# 0, 1 and 2 in a single pass
+red = 0 # In this case this is the first color of the flag.
+white = 1 # This is the second color of the flag.
+blue = 2 # This is the third color of the flag.
+colors = [red, white, blue]
+# Function to sort array
+def dnf_sort( sequence:list, arr_size:int)->list:
```

Comment:
```suggestion
red = 0  # In this case this is the first color of the flag.
white = 1  # This is the second color of the flag.
blue = 2  # This is the third color of the flag.
colors = (red, white, blue)
# Function to sort array
def dutch_national_flag_sort(sequence: list, seq_size: int) -> list:
```

Reply:
Why do we ask for the size of `sequence` if we can obtain is by doing `len(sequence)`?  What happens if the caller says it is 3 but it is 5?  What happens if the caller says it is 7 but it is 5?

Reply:
@cclauss  yes  i have updated seq_size.please review and confirm